### PR TITLE
Implement a Redis StatefulSet Use Case

### PR DESCRIPTION
**Pull request by yudaykiran**
**What this PR does / why we need it**:
- Add a YAML for Redis StatefulSet and Headless Service.
- Create ReadME.md to consisting steps for deployment and use-case demonstration.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
Fix for Issue #852 

**Special notes for your reviewer**:
Signed-off-by: yudaykiran <udaykiran.y@gmail.com>

### CODE REVIEW

File: k8s/demo/redis/README.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+Get the status of running StatefulSets:
+
+```bash
+ubuntu@kubemaster:~$ kubectl get statefulset
+NAME      DESIRED   CURRENT   AGE
+rd        3         3         19h
+
+```
+
+Get the status of underlying persistent volume being used by Redis StatefulSet:
```
Comment (mahebbar) [inline k8s/demo/redis/README.md]: volume being used = volume used by

File: k8s/demo/redis/README.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+kubernetes                                          ClusterIP   10.96.0.1        <none>        443/TCP             6d
+maya-apiserver-service                              ClusterIP   10.111.26.252    <none>        5656/TCP            6d
+pvc-1f305192-ca11-11e7-892e-000c29119159-ctrl-svc   ClusterIP   10.105.218.103   <none>        3260/TCP,9501/TCP   19h
+pvc-59eea5e9-ca11-11e7-892e-000c29119159-ctrl-svc   ClusterIP   10.106.116.112   <none>        3260/TCP,9501/TCP   19h
+pvc-e7b2a235-ca10-11e7-892e-000c29119159-ctrl-svc   ClusterIP   10.102.32.23     <none>        3260/TCP,9501/TCP   19h
+redis                                               ClusterIP   None             <none>        6379/TCP            19h
+
+```
+
+## Check Redis replication
```
Comment (mahebbar) [inline k8s/demo/redis/README.md]: replication = Replication

File: k8s/demo/redis/README.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+Set a key:value pair in the redis master.
```
Comment (mahebbar) [inline k8s/demo/redis/README.md]: redis = Redis

File: k8s/demo/redis/README.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+```bash
+ubuntu@kubemaster:~kubectl exec rd-0 -- /opt/redis/redis-cli -h rd-0.redis SET replicated:test true
+OK
+
+```
+
+Retrieve the value of the key from a redis slave.
```
Comment (mahebbar) [inline k8s/demo/redis/README.md]: redis =] Redis

**Comment by yudaykiran** [comment]
@mahebbar Implemented review comments.

**Pull request merged**